### ISSUE TITLE
Some applications will not use ruby version specified in Gemfile and fail to build, while others are fine

### ISSUE THREAD

**Issue by thekeith** [aws-amplify/amplify-console#1146]
I have two separate Jekyll applications. Each has exactly the same Gemfile and use the same ruby version, with the Gemfile.lock also being the same. Each provisioned instance is also exactly the same as per the provision output logs.

App A builds without issue, with the bundler version on the docker instance being 2.1.2 even though the Gemfile calls for 2.1.4
App B fails to build, with bundler on the docker instance being 2.0.4. The build fails due to ruby being 2.4.6 even though the Gemfile specified 2.7.0

Despite all configuration variables (build settings, etc.) being exactly the same, one fails while the other succeeds.

The error output is as follows:

```
                                 # Starting phase: preBuild
                                 # Executing command: bundle install
2020-10-17T19:34:46.533Z [WARNING]: Don't run Bundler as root. Bundler can ask for sudo if it is needed, and
                                    installing your bundle as root will break this application for all non-root
                                    users on this machine.
2020-10-17T19:34:46.970Z [WARNING]: Warning: the running version of Bundler (2.0.1) is older than the version that created the lockfile (2.1.4). We suggest you upgrade to the latest version of Bundler by running `gem install bundler`.
2020-10-17T19:34:46.978Z [WARNING]: Your Ruby version is 2.4.6, but your Gemfile specified 2.7.0
2020-10-17T19:34:46.983Z [ERROR]: !!! Build failed
2020-10-17T19:34:46.983Z [ERROR]: !!! Non-Zero Exit Code detected
2020-10-17T19:34:46.984Z [INFO]: # Starting environment caching...
2020-10-17T19:34:46.984Z [INFO]: # Environment caching completed
Terminating logging...
```

**Comment by thekeith** [comment]
Note: Changing ruby versions in the Gemfile allowed it to successfully build, but the issue remains that two functionally identical apps are not able to both be built.

**Comment by litwicki** [comment]
@thekeith if I'm understanding correctly your build depends on a version of ruby that is not the standard part of the build container. When true, you'll need to do exactly as you said here where updating the Gemfile will properly define the version to use for the build and install it to the container.

**Comment by matschaffer** [comment]
I came across this via the first error (bundler version). I worked around that by adding `gem install bundler --version $(tail -n1 Gemfile.lock)` to my build script, but looks like the ruby in the docker image is too old (2.4.6) to run the version of active support I picked up via jekyll-mentions.

This is a site which I was originally planning to use github pages for but decided to try amplify after finding that github pages can't handle both apex and www at the same time.
